internal/cmd/testenv/main.go

```diff
@@ -0,0 +1,189 @@
+package main
+
+import (
+	"context"
+	"errors"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+
+	"github.com/kr/pretty"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/internal/testsetup"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/aws"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/azure"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/gcp"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql"
+	"github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/graphql/core"
+	polaris_log "github.com/rubrikinc/rubrik-polaris-sdk-for-go/pkg/polaris/log"
+	"golang.org/x/sync/errgroup"
+)
+
+func main() {
+	checkOnly := func() bool {
+		cleanup := flag.Bool("cleanup", false, "Perform cleanup tasks post CI")
+		precheck := flag.Bool("precheck", false, "Check pre-requirements for running CI tests")
+		flag.Parse()
+		if *cleanup == *precheck {
+			flag.Usage()
+			os.Exit(1)
+		}
+		return *precheck
+	}()
```

Comment:
Smaller thing, since we have separated out the `precheck` and `cleanup` steps to their own functions we don't really need a function for this. We could just use `*precheck` below where we use `checkOnly` now.